src/app/(pages)/educations/_components/List/EducationsListTable.tsx

```diff
@@ -6,36 +6,68 @@ import {
   TableHeader,
   TableRow,
 } from "~/components/organisms/Table";
-import { Pencil } from "lucide-react";
 import PaginationComponent from "~/components/organisms/pagination/PaginationComponent";
-import { type FC } from "react";
+import { useState, useEffect, type FC } from "react";
 import type { FindEducationReturnDTO } from "~/server/services/education/education.repository";
 import { translateEducationType } from "~/app/(pages)/educations/utils";
 import { type EducationType } from "~/server/db/schema";
+import Link from "next/link";
+import { api } from "~/trpc/react";
+import { Trash2, Pencil } from "lucide-react";
 
 type Props = {
   data?: FindEducationReturnDTO[];
   totalPageNumber: number;
 };
 
-const EducationsListTable: FC<Props> = ({ data, totalPageNumber }) => {
+const EducationsListTable: FC<Props> = ({
+  data: initialData,
+  totalPageNumber,
+}) => {
+  const [data, setData] = useState<FindEducationReturnDTO[]>([]);
+
+  useEffect(() => {
+    if (initialData) {
+      setData(initialData);
+    }
+  }, [initialData]);
+
+  const deleteEducation = api.education.deleteById.useMutation();
+
+  const handleDelete = async (id: string) => {
+    const confirmed = window.confirm(
+      "Jeste li sigurni da želite obrisati ovu edukaciju?",
+    );
+    if (!confirmed) return;
+
+    try {
+      await deleteEducation.mutateAsync({ id });
+      // Update the state to reflect the deletion
+      setData((prevData) =>
```

Comment:
It is always better to refetch items from database than to manipulate the memory when updating/creating/deleting items. It is not that "expensive" but you are 100% sure that data you are displaying is correct. So here you can set filter to empty or set page to 0 which is going to re-trigger the process of fetching data and rendering them.

Reply:
Setting it like this doesn't do anything
```ts
      setData((prevData) =>
        prevData?.filter(() => []),
      );
```

Reply:
Sorry I think you've misunderstood me. 
I thought that you should not delete the item using tRPC request and on successful response remove that item in the memory of the client and than set the state. Instead of data manipulation you should re-fetch the data from the server and set new state.

Reply:
I've committed the change. Did you mean it the way I did it now?

Reply:
It looks good.

Just another thing. There is now no use for the `useState` and `useEffect` because data is provided from the parent component